test/functional/legacy/eval_spec.lua

```diff
@@ -495,6 +495,18 @@ describe('eval', function()
       ' abcE4b10-4\000abcE4b10-4-2')
   end)
 
+  it("setreg('0',getreg('a',1,1)) doesn't crash for unset register", function()
```

Comment:
well getreg() _sometimes_ is mutable and sometimes not is bad interface, bram agreed that much. This PR is  the neovim resolution of that, as neovim are not as afraid of "excessive" memory allocation as vim, so we don't need tricks to fake the mutability of a NULL list and can just alloc it. I will add tests for mutability here.